### PR TITLE
LPD-27773 Taken into account ctCollectionId during DDMValidation upgrade process of the DDMStructure and DDMStructureVersion entities

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPD-27773

### CODE REVIEW

**Comment by p-albuquerque** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_1/DDMValidationUpgradeProcess.java [hunk 1]
```diff
@@ -121,7 +121,9 @@ private void _upgradeDDMStructure(
 
 					long structureId = resultSet.getLong("structureId");
 
-					updatePreparedStatement1.setLong(2, structureId);
+					updatePreparedStatement1.setLong(
+						2, resultSet.getLong("ctCollectionId"));
+					updatePreparedStatement1.setLong(3, structureId);
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_1/DDMValidationUpgradeProcess.java:126]: @p-albuquerque declare the variable as close as possible to its first usage

```
updatePreparedStatement1.setString(
	1, definitionJSONObject.toString());
updatePreparedStatement1.setLong(
	2, resultSet.getLong("ctCollectionId"));

long structureId = resultSet.getLong("structureId");

updatePreparedStatement1.setLong(3, structureId);
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_1_1/test/DDMValidationUpgradeProcessDDMStructureTest.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Paulo Albuquerque
+ */
+@RunWith(Arquillian.class)
+public class DDMValidationUpgradeProcessDDMStructureTest
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_1_1/test/DDMValidationUpgradeProcessDDMStructureTest.java:48]: @p-albuquerque in my opinion, the class name should have the suffix `UpgradeProcessTest` to follow the existing pattern in the portal. 

Let's try to rename it to `DDMValidationStructureUpgradeProcessTest`

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_1_1/test/DDMValidationUpgradeProcessDDMStructureTest.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+		ddmFormField.setDDMFormFieldValidation(ddmFormFieldValidation);
+
+		DDMForm ddmForm = DDMFormTestUtil.createDDMForm();
+
+		List<DDMFormField> ddmFormFields = ddmForm.getDDMFormFields();
+
+		ddmFormField.setDDMForm(ddmForm);
+
+		ddmFormFields.add(ddmFormField);
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_1_1/test/DDMValidationUpgradeProcessDDMStructureTest.java:112]: @p-albuquerque why don't you just call the `ddmForm.addDDMFormField(ddmFormField);` here?

Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_1_1/test/DDMValidationUpgradeProcessDDMStructureTest.java:112]: @p-albuquerque reorder:

```
DDMForm ddmForm = DDMFormTestUtil.createDDMForm();

DDMFormField ddmFormField = DDMFormTestUtil.createDDMFormField(
		RandomTestUtil.randomString(), RandomTestUtil.randomString(),
		DDMFormFieldTypeConstants.DATE, DDMFormFieldTypeConstants.DATE,
		false, false, false);

DDMFormFieldValidation ddmFormFieldValidation =
	new DDMFormFieldValidation();

ddmFormFieldValidation.setDDMFormFieldValidationExpression(
	new DDMFormFieldValidationExpression() {
		{
			setName("dateRange");
			setValue("dateRange({parameter} AND {parameter})");
		}
	});

ddmFormField.setDDMFormFieldValidation(ddmFormFieldValidation);

ddmForm.addDDMFormField(ddmFormField);
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_1_1/test/DDMValidationUpgradeProcessDDMStructureTest.java (hunk 2 above)
Comment (p-albuquerque) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_1_1/test/DDMValidationUpgradeProcessDDMStructureTest.java:48]: After discussion with @marcelabc We propose the following:
**DDMStructure**DDMValidationUpgradeProcessTest
**DDMStructureVersion**DDMValidationUpgradeProcessTest

What do you think, @carolmariaabb ?

Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_1_1/test/DDMValidationUpgradeProcessDDMStructureTest.java:48]: @p-albuquerque ok!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_1_1/test/DDMValidationUpgradeProcessDDMStructureTest.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+	@Override
+	protected CTModel<?> addCTModel() throws Exception {
+		_ddmStructure = DDMStructureTestUtil.addStructure(
+			_group.getGroupId(), DDMFormInstance.class.getName(),
+			DDMFormTestUtil.createDDMForm(RandomTestUtil.randomString()));
+
+		setDefinition(_ddmStructure);
+
+		return _ddmStructureLocalService.updateDDMStructure(_ddmStructure);
```
[the remark this answers is not in the document]
Comment (p-albuquerque) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_1_1/test/DDMValidationUpgradeProcessDDMStructureTest.java]: @carolmariaabb Do you have any thoughts about this situation?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_1/DDMValidationUpgradeProcess.java [hunk 5]
```diff
@@ -119,9 +119,12 @@ private void _upgradeDDMStructure(
 					updatePreparedStatement1.setString(
 						1, definitionJSONObject.toString());
 
+					updatePreparedStatement1.setLong(
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_1/DDMValidationUpgradeProcess.java:122]: @p-albuquerque remove the empty line

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_1_1/test/DDMStructureDDMValidationUpgradeProcessTest.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+				{
+					setName("dateRange");
+					setValue("dateRange({parameter} AND {parameter})");
+				}
+			});
+
+		DDMFormField ddmFormField = DDMFormTestUtil.createDDMFormField(
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			DDMFormFieldTypeConstants.DATE, DDMFormFieldTypeConstants.DATE,
+			false, false, false);
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_1_1/test/DDMStructureDDMValidationUpgradeProcessTest.java:101]: @p-albuquerque declare `DDMFormField ddmFormField` before `DDMFormFieldValidation ddmFormFieldValidation`

see my previous [comment](https://github.com/liferay-forms/liferay-portal/pull/2574#discussion_r1695129263)

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_1_1/test/DDMValidationUpgradeProcessDDMStructureTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_1_1/test/DDMValidationUpgradeProcessDDMStructureTest.java]: @p-albuquerque you can keep this way. I was just trying to understand 😄

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:forward

**Pull request closed without merging**